src/commands/notify.rs

```diff
@@ -0,0 +1,710 @@
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "{}", self.msg.display_dig_style())?;
+        writeln!(f, ";; Query time: {} msec", self.time.num_milliseconds())?;
+        if let Some(server) = self.server {
+            writeln!(f, ";; Server: {}#{}", server.ip(), server.port())?;
+        }
+        writeln!(
+            f,
+            ";; WHEN: {}",
+            self.when.format("%a %b %d %H:%M:%S %Z %Y")
+        )?;
+        writeln!(f, ";; MSG SIZE  rcvd: {}", self.msg.as_slice().len())?;
+        Ok(())
+    }
+}
+
+impl FromStr for TSigInfo {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let Some((mut name, rest)) = s.split_once(':') else {
+            return Err("should contain at least one `:`".into());
+        };
+
+        let mut key;
+        let mut algorithm;
+        if let Some((k, a)) = rest.split_once(':') {
+            key = k;
+            algorithm = a;
+        } else {
+            key = rest;
+            // This is different from ldns's default of
+            // hmac-md5.sig-alg.reg.int but we don't support that algorithm.
+            algorithm = "hmac-sha512";
+        }
+
+        // With dig TSIG keys are also specified with -y,
+        // but our format is: <name:key[:algo]>
+        //      and dig's is: [hmac:]name:key
+        //
+        // When we detect an unknown tsig algorithm in algo,
+        // but a known algorithm in name, we can assume dig
+        // order was used.
+        //
+        // We can correct this by checking whether the name contains a valid
+        // algorithm while the algorithm doesn't.
+        if Algorithm::from_str(algorithm).is_err() && Algorithm::from_str(name).is_ok() {
+            (name, key, algorithm) = (key, algorithm, name);
+        }
+
+        let algorithm = Algorithm::from_str(algorithm)
+            .map_err(|_| format!("Unsupported TSIG algorithm: {algorithm}"))?;
+
+        let key = base64::decode(key).map_err(|e| format!("TSIG key is invalid base64: {e}"))?;
+
+        let name =
+            Name::<Array<255>>::from_str(name).map_err(|e| format!("TSIG name is invalid: {e}"))?;
+
+        Ok(TSigInfo {
+            name,
+            key,
+            algorithm,
+        })
+    }
+}
+
+#[derive(Clone, Debug, clap::Args, PartialEq, Eq)]
+pub struct Notify {
+    /// The zone
+    #[arg(short = 'z', long = "zone", required = true)]
+    zone: Name<Vec<u8>>,
+
+    // The -I option is supported by ldns but is not available in domain yet.
+    // It requires creating a connection from a UdpSocket (or similar).
+    // /// Source address to query from
+    // #[arg(short = 'I', required = false)]
+    // source_address: (),
+    //
+    /// SOA version number to include
+    #[arg(short = 's')]
+    soa_version: Option<u32>,
+
+    /// A base64 tsig key and optional algorithm to include
+    #[arg(
+        short = 'y',
```

Comment:
Why is this `-y`?  Do we want to change it to something more obvious, or should we keep it as-is for ldns compatibility?

Reply:
FYI other tools also use `-y`, e.g. `dig`.